File: 0242-valid-anagram/0242-valid-anagram.cs
public class Solution {
    public bool IsAnagram(string s, string t) {
        Dictionary<char, int> dict = new Dictionary<char, int>();
        foreach (char c in s) { 
            dict[c] = 1 + dict.GetValueOrDefault(c, 0) ; 
        }
        foreach (char c in t) { 
            if (!dict.ContainsKey(c)) return false; 
            dict[c] = dict[c] -1 ;
        }

        foreach (var kvp in dict) { 
            if (kvp.Value !=0) { 
                return false;
            }
        }

        return true;
       


        
    }
}

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        hmap = defaultdict(int) 
        for i in s : 
            hmap[i]+=1 
        for i in t : 
            if i in hmap : 
                hmap[i]-=1
            else : 
                return False 
        for v in hmap.values() : 
            if v != 0 : 
                return False 
        return True
        